### ISSUE TITLE
usage of moved variable in same function call not detected by clang-tidy when function takes rvalue

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#55388]
```cpp
#include <list>

extern void func(std::list<int>&&, std::list<int>); // remove "&&" for the warning to appear
extern std::list<int> func2(const std::list<int>&);

static void f(std::list<int>l)
{
    func(std::move(l), func2(l));
    l.size(); // bugprone-use-after-move detected
}
```

https://godbolt.org/z/oYEP37nYc

**Comment by sh7dm** [comment]
I'm currently on it, understood the issue behind and trying to fix AST selectors. Might still need help with needed behavior of those selectors and implementing  the fix.

**Comment by sh7dm** [comment]
This looks like intended behavior: https://github.com/llvm/llvm-project/blob/1584b2c74e4c804a2c85d760a1a2c10b33465f2e/clang-tools-extra/test/clang-tidy/checkers/bugprone-use-after-move.cpp#L1004 . The point behind that is that this pattern does not cause value to be used after move, since move happens in the function call, which happens after argument evaluation.

**Comment by firewave** [comment]
> This looks like intended behavior:
> 
> https://github.com/llvm/llvm-project/blob/1584b2c74e4c804a2c85d760a1a2c10b33465f2e/clang-tools-extra/test/clang-tidy/checkers/bugprone-use-after-move.cpp#L1004
> 
> . The point behind that is that this pattern does not cause value to be used after move, since move happens in the function call, which happens after argument evaluation.

I have no knowledge about this but #56748 suggests it should not be intended since the order is undefined.

**Comment by sh7dm** [comment]
This is about the meaning of `&&`, which will make move happen definitely in the callee.

**Comment by PiotrZSL** [comment]
Actually it looks to be correct:
```
/fpwork/private1/1.cpp:9:5: warning: 'l' used after it was moved [bugprone-use-after-move]
    l.size(); // bugprone-use-after-move detected
    ^
/fpwork/private1/1.cpp:8:5: note: move occurred here
    func(std::move(l), func2(l));
```
Warning is reported only for `l.size()` simply because when func2 were called, the move on l were not yet executed, that was just a static cast.
